components/brave_vpn/brave_vpn_os_connection_api.cc

```diff
@@ -402,59 +398,19 @@ void BraveVPNOSConnectionAPI::ParseAndCacheHostnames(
           << hostname_->display_name << ", " << hostname_->is_offline << ", "
           << hostname_->capacity_score;
 
-  if (skus_credential_.empty()) {
-    VLOG(2) << __func__ << " : skus_credential is empty";
-    UpdateAndNotifyConnectionStateChange(ConnectionState::CONNECT_FAILED);
-    return;
-  }
-
   if (!GetAPIRequest()) {
     CHECK_IS_TEST();
     return;
   }
 
-  // Get subscriber credentials and then get EAP credentials with it to create
-  // OS VPN entry.
-  VLOG(2) << __func__ << " : request subscriber credential:"
+  // Get profile credentials it to create OS VPN entry.
+  VLOG(2) << __func__ << " : request profile credential:"
           << GetBraveVPNPaymentsEnv(GetCurrentEnvironment());
-  GetAPIRequest()->GetSubscriberCredentialV12(
-      base::BindOnce(&BraveVPNOSConnectionAPI::OnGetSubscriberCredentialV12,
-                     base::Unretained(this)),
-      skus_credential_, GetBraveVPNPaymentsEnv(GetCurrentEnvironment()));
-}
-
-void BraveVPNOSConnectionAPI::OnGetSubscriberCredentialV12(
-    const std::string& subscriber_credential,
-    bool success) {
-  if (cancel_connecting_) {
-    UpdateAndNotifyConnectionStateChange(ConnectionState::DISCONNECTED);
-    cancel_connecting_ = false;
-    return;
-  }
-
-  if (!success) {
-    VLOG(2) << __func__ << " : failed to get subscriber credential";
-    if (subscriber_credential == kTokenNoLongerValid) {
-      for (auto& obs : observers_)
-        obs.OnGetInvalidToken();
-    }
-    UpdateAndNotifyConnectionStateChange(ConnectionState::CONNECT_FAILED);
-    return;
-  }
-
-  VLOG(2) << __func__ << " : received subscriber credential";
-
-  if (!GetAPIRequest()) {
-    CHECK_IS_TEST();
-    return;
-  }
 
-  // TODO(bsclifton): consider storing `subscriber_credential` for
-  // support ticket use-case (see `CreateSupportTicket`).
```

Comment:
It's a little bit outside the scope here - but we could add this credential now to [`BraveVpnAPIRequest::CreateSupportTicket`](https://github.com/brave/brave-core/blob/37259e4189f1d68afc482421bd1e0df6ca1c678e/components/brave_vpn/brave_vpn_api_request.cc#L216-L219) (adding `subscriber_credential` as a 5th param) and then to the [`GetValueWithTicketInfos` util function](https://github.com/brave/brave-core/blob/3bed8ea4fe4dab2a61a4d418cc6f797def00bfdb/components/brave_vpn/brave_vpn_service_helper.cc#L158-L173) (and then call from `BraveVpnService`)

I can create a follow up ticket for that if you prefer 😄

Reply:
Resolved!

Reply:
So awesome- thank you! 🙂 This should help provide better support